Tests hang on Apple Silicon during cpu feature detection

Apple Silicon (M2) MacBook Pro, MacOS Ventura 13.2.1, Xcode-14.2, Python-3.11.2.

Freshly-cloned repo.

Installed what's listed in the `requirements.txt`. 

Basically, `pytest` gets to
```
. . .
rootdir: /Users/ur20980/src/PQClean/test, configfile: pytest.ini
plugins: xdist-3.2.1, forked-1.3.0
[gw0] darwin Python 3.11.2 cwd: /Users/ur20980/src/PQClean/test
[gw0] Python 3.11.2 (main, Feb 24 2023, 04:00:33) [Clang 14.0.0 (clang-1400.0.29.202)]
gw0 ok
```
and hangs up. The only way to move from that state is "Ctrl-C" (several times).


I'm passing on the stack trace from the "kill", perhaps it could help you figure what's wrong:
```
$ cd test
$ pytest -v -n=1
======================================== test session starts =========================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/ur20980/src/PQClean/test, configfile: pytest.ini
plugins: xdist-3.2.1, forked-1.3.0
[gw0] darwin Python 3.11.2 cwd: /Users/ur20980/src/PQClean/test
[gw0] Python 3.11.2 (main, Feb 24 2023, 04:00:33) [Clang 14.0.0 (clang-1400.0.29.202)]
gw0 ok^C^C!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py:324: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
Traceback (most recent call last):
  File "/opt/local/bin/pytest", line 8, in <module>
    sys.exit(console_main())
             ^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_pytest/config/__init__.py", line 190, in console_main
    code = main()
           ^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_pytest/config/__init__.py", line 167, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_pytest/main.py", line 317, in pytest_cmdline_main
    return wrap_session(config, _main)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_pytest/main.py", line 305, in wrap_session
    config.hook.pytest_sessionfinish(
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_callers.py", line 55, in _multicall
    gen.send(outcome)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/_pytest/terminal.py", line 808, in pytest_sessionfinish
    outcome.get_result()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/xdist/dsession.py", line 87, in pytest_sessionfinish
    nm.teardown_nodes()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/xdist/workermanage.py", line 81, in teardown_nodes
    self.group.terminate(self.EXIT_TIMEOUT)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/execnet/multi.py", line 215, in terminate
    safe_terminate(
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/execnet/multi.py", line 311, in safe_terminate
    reply.get()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/execnet/gateway_base.py", line 206, in get
    self.waitfinish(timeout)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/execnet/gateway_base.py", line 213, in waitfinish
    if not self._result_ready.wait(timeout):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 622, in wait
    signaled = self._cond.wait(timeout)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 320, in wait
    waiter.acquire()
KeyboardInterrupt

$ Cleaning up testcases directory
```

Overall, I'd call #484 a good step towards success. But the code seems to attempt to use AVX2 implementations at least in some places - that can't work on AARCH64:
```text
$ time pytest -v -n=auto
======================================== test session starts =========================================
platform darwin -- Python 3.11.3, pytest-7.2.2, pluggy-1.0.0 -- /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/ur20980/src/PQClean/test, configfile: pytest.ini
plugins: xdist-3.2.1, forked-1.3.0
[gw0] darwin Python 3.11.3 cwd: /Users/ur20980/src/PQClean/test                                
[gw1] darwin Python 3.11.3 cwd: /Users/ur20980/src/PQClean/test  
.  .  .
gw0 [4361] / gw1 [4361] / gw2 [4361] / gw3 [4361] / gw4 [4361] / gw5 [4361] / gw6 [4361] / gw7 [4361] gw0 [4361] / gw1 [4361] / gw2 [4361] / gw3 [4361] / gw4 [4361] / gw5 [4361] / gw6 [4361] / gw7 [4361] / gw8 [4361] / gw9 [4361] / gw10 [4361] / gw11 [4361]
scheduling tests via LoadScheduling

test_char.py::test_char[sse implementation of mceliece6688128f] 
test_api_h.py::test_api_h[avx2 implementation of sphincs-sha2-256s-robust] 
[gw1] [  0%] PASSED test_api_h.py::test_api_h[avx2 implementation of sphincs-sha2-256s-robust] 
test_api_h.py::test_api_h[clean implementation of sphincs-sha2-256s-robust] 
test_char.py::test_char[clean implementation of dilithium2] 
test_duplicate_consistency.py::test_duplicate_consistency[duplicate_consistency/mceliece8192128f_sse.yml: mceliece8192128f sse18] 
[gw7] [  0%] PASSED test_duplicate_consistency.py::test_duplicate_consistency[duplicate_consistency/mceliece8192128f_sse.yml: mceliece8192128f sse18] 
test_duplicate_consistency.py::test_duplicate_consistency[duplicate_consistency/kyber512_aarch64.yml: kyber512 aarch640]
.  .  .
[gw8] [ 41%] PASSED test_duplicate_consistency.py::test_duplicate_consistency[duplicate_consistency/sphincs-sha2-128f-robust_avx2.yml: sphincs-sha2-128f-robust avx22] 
test_format.py::test_format[avx2 implementation of sphincs-shake-128s-simple] 
test_duplicate_consistency.py::test_duplicate_consistency[duplicate_consistency/dilithium5aes_clean.yml: dilithium5aes clean6] 
[gw0] [ 41%] SKIPPED test_format.py::test_format[clean implementation of sphincs-sha2-128f-robust] 
test_functest.py::test_functest[clean implementation of kyber768] 
[gw8] [ 41%] SKIPPED test_format.py::test_format[avx2 implementation of sphincs-shake-128s-simple] 
test_format.py::test_format[clean implementation of sphincs-shake-128s-simple]
.  .  .
[gw2] [ 54%] PASSED test_boolean.py::test_boolean[clean implementation of kyber768-90s] 
test_boolean.py::test_boolean[sse implementation of mceliece6688128f] 
test_char.py::test_char[avx2 implementation of sphincs-shake-192s-simple] 
[gw8] [ 54%] FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-192s-robust] 
test_functest.py::test_functest[clean implementation of sphincs-shake-256s-simple] 
[gw4] [ 54%] PASSED test_char.py::test_char[vec implementation of mceliece460896] 
test_char.py::test_char[avx implementation of mceliece460896] 
.  .  .
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
===================== 345 failed, 3590 passed, 426 skipped in 447.66s (0:07:27) ===
```

Some of the examples of failures:
```text
__________ test_makefile_dependencies[aarch64 implementation of sphincs-shake-256s-simple] ___________
[gw11] darwin -- Python 3.11.3 /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11

implementation = <Implementation(sphincs-shake-256s-simple, aarch64)>
impl_path = '/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eEl1PMP9/crypto_bla/scheme/impl'
test_dir = '/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eEl1PMP9'
init = <function isolate_test_files.<locals>.initializer at 0x1100a67a0>
destr = <function isolate_test_files.<locals>.destructor at 0x1100a6840>

    @pytest.mark.parametrize(
        'implementation,test_dir,impl_path, init, destr',
        [(impl,
          *helpers.isolate_test_files(impl.path(), 'test_makefile_deps_'))
         for impl in pqclean.Scheme.all_supported_implementations()],
        ids=[str(impl) for impl in pqclean.Scheme.all_supported_implementations()],
    )
    @helpers.filtered_test
    def test_makefile_dependencies(implementation, impl_path, test_dir,
                                   init, destr):
        init()
        # initial build - want to have *all* files in place at beginning
        helpers.make('clean', working_dir=impl_path)
>       helpers.make(working_dir=impl_path)

test_makefile_dependencies.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers.py:135: in make
    return run_subprocess(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['make']
working_dir = '/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eEl1PMP9/crypto_bla/scheme/impl'
env = None, expected_returncode = 0, print_output = True

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0,
                       print_output=True):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        if env is not None:
            env_ = os.environ.copy()
            env_.update(env)
            env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which the unittest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
        result = subprocess.run(
            command,
            stdout=subprocess.PIPE,
            stderr=subprocess.STDOUT,
            cwd=working_dir,
            env=env,
        )
        if print_output:
            print(result.stdout.decode('utf-8'))
        if expected_returncode is not None:
>           assert result.returncode == expected_returncode, \
                "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code 2

helpers.py:110: AssertionError
---------------------------------------- Captured stdout call ----------------------------------------
/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eEl1PMP9/crypto_bla/scheme/impl > make clean
rm -f address.o context_shake.o f1600x2.o f1600x2_const.o fips202x2.o fors.o hash_shake.o hash_shakex2.o merkle.o sign.o thash_shake_simplex2.o utils.o utilsx2.o wots.o
rm -f libsphincs-shake-256s-simple_aarch64.a

/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eEl1PMP9/crypto_bla/scheme/impl > make
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o address.o address.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o context_shake.o context_shake.c
as -o f1600x2.o f1600x2.s
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o f1600x2_const.o f1600x2_const.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o fips202x2.o fips202x2.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o fors.o fors.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o hash_shake.o hash_shake.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o hash_shakex2.o hash_shakex2.c
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o merkle.o merkle.c
merkle.c:63:17: error: implicit conversion changes signedness: 'int' to 'uint32_t' (aka 'unsigned int') [-Werror,-Wsign-conversion]
                ~0 /* ~0 means "don't bother generating an auth path */ );
                ^~
1 error generated.
make: *** [merkle.o] Error 1
```

And
```text
======= short test summary info =====
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of kyber768] - AssertionError: Clang-tidy returned 1
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of dilithium3] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Clang-tidy returned 1
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of kyber512] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of dilithium2] - AssertionError: Clang-tidy returned 1
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of dilithium5] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of kyber1024] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Clang-tidy returned 1
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Clang-tidy returned 1
FAILED test_dynamic_memory.py::test_dynamic_memory[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Clang-tidy returned 1
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Clang-tidy returned 1
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_compile_lib.py::test_compile_lib[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber768] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber1024-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber768] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of kyber768] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium5] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber512] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of dilithium5] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of hqc-rmrs-192] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber512] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of hqc-rmrs-192] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of kyber512] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium3] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber512-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber1024] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of hqc-rmrs-256] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_linter.py::test_clang_tidy[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Clang-tidy returned 1
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium3] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber512-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber1024] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of hqc-rmrs-256] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of kyber1024] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of dilithium3] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of falcon-512] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium2aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of falcon-512] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber768-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium2aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of kyber768-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of falcon-1024] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of falcon-1024] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[vec implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium2] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of hqc-rmrs-128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[sse implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium2] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of kyber1024-90s] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of hqc-rmrs-128] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of dilithium2] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium3aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium5] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium3aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of falcon-1024] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium5aes] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of dilithium3] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of dilithium5aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of dilithium5aes] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[aesni implementation of sphincs-haraka-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-haraka-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[avx2 implementation of sphincs-sha2-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_metadata_sizes.py::test_metadata_sizes[clean implementation of sphincs-sha2-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of falcon-512] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece348864] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber768] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium3aes] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber1024-90s] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of kyber768] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium2] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece6960119] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of dilithium2] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece8192128f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber512] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium5] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium2aes] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber512-90s] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of dilithium5] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of kyber512] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece8192128] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece348864f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_symbol_namespace.py::test_symbol_namespaces[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber768-90s] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece6688128f] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece6960119f] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of hqc-rmrs-192] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of kyber1024] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of kyber1024] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-sha2-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of dilithium3] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece460896] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-192f-robust] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece460896f] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-256f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of hqc-rmrs-128] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[vec implementation of mceliece6688128] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of hqc-rmrs-256] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[aarch64 implementation of sphincs-shake-128f-simple] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_nistkat.py::test_nistkat[clean implementation of sphincs-haraka-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-128s-robust] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-192s-simple] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_testvectors.py::test_testvectors[aarch64 implementation of sphincs-shake-256s-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-256f-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-192f-simple] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-128f-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_functest.py::test_functest[aarch64 implementation of sphincs-shake-128s-simple] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-192s-robust] - AssertionError: Got unexpected return code 2
FAILED test_makefile_dependencies.py::test_makefile_dependencies[aarch64 implementation of sphincs-shake-256s-simple] - AssertionError: Got unexpected return code 2
```
Most of the above tests should not have been even attempted (avx2, sse, etc.).

What would happen if you try to force Mx chips (M1 and M2) as ARM64 or aarch64?

Thankfully, the current master stopped hanging on tests on Apple Silicon.

However, 122 tests keep failing, with errors such as:
```text
clang -std=c99 -O3 -Wall -Wconversion -Werror -Wextra -Wmissing-prototypes -Wpedantic -Wredundant-decls -I../../../common  -c -o utilsx2.o utilsx2.c
utilsx2.c:91:58: error: implicit conversion changes signedness: 'int' to 'unsigned int' [-Werror,-Wsign-conversion]
            if ((((internal_idx << 1) ^ internal_leaf) & ~0x1) == 0) {
                                                       ~ ^~~~
utilsx2.c:116:65: error: implicit conversion changes signedness: 'int' to 'unsigned int' [-Werror,-Wsign-conversion]
                               (2 / 2) * (internal_idx & ~1U) + j - left_adj + internal_idx_offset );
                                                              ~ ^
2 errors generated.
make: *** [utilsx2.o] Error 1
```
and
```text
command = ['make']
working_dir = '/Users/ur20980/src/PQClean/testcases/test_makefile_deps__eMDIqFIe/crypto_bla/scheme/impl'
env = None, expected_returncode = 0, print_output = True

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0,
                       print_output=True):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        if env is not None:
            env_ = os.environ.copy()
            env_.update(env)
            env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which the unittest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
        result = subprocess.run(
            command,
            stdout=subprocess.PIPE,
            stderr=subprocess.STDOUT,
            cwd=working_dir,
            env=env,
        )
        if print_output:
            print(result.stdout.decode('utf-8'))
        if expected_returncode is not None:
>           assert result.returncode == expected_returncode, \
                "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code 2

helpers.py:110: AssertionError
```

Complete log of `pytest -v -n=auto`:
[pqclean-tests.out.txt](https://github.com/PQClean/PQClean/files/11481437/pqclean-tests.out.txt)

--- a/test/helpers.py
+++ b/test/helpers.py
@@ -281,5 +281,15 @@ def get_cpu_info():
     while the_info is None or 'flags' not in the_info:
         import cpuinfo
         the_info = cpuinfo.get_cpu_info()
+        if 'Apple M' in the_info.get('brand_raw', ''):
+            the_info['flags'] = [
+                'sha3',
+                'sha256',
+                'sha1',
+                'aes',
+                'crc32',
+                'neon',
+                'asimd',
+            ]
 
     return the_info

